### ISSUE TITLE
BatchNorm gets inconsistent result between Module API and NDArray API 

### ISSUE THREAD

**Issue by sui-hs** [apache/incubator-mxnet#14710]
## Description
when using BatchNorm layer in NN, I found that the results differ in Module(Symbol) API and NDArray API. 

## Environment info (Required)
MxNet version: 1.3.0

## Error Message:
1. [with this script](https://gist.github.com/sui-hs/ec86ef6622e8363908ebe4202a335567) the BatchNorm result is:
[[-1.1049458   0.45291406]
 [ 1.3138113  -1.3863459 ]
 [-0.20886569  0.9334318 ]]

2. [with this script](https://gist.github.com/sui-hs/b36ee22028324b8a6fef6416033e48ed), I got:
[[-34.330654   31.538832 ]
 [  8.9485655 -47.63322  ]
 [-18.296946   52.223007 ]]

## Steps to reproduce
(Paste the commands you ran that produced the error.)

1. run the first script
2. run the second script

## What have you tried to solve it?

1. I tried Executor API directly with the first model, and got the second result.
2. I tried with TensorFlow, and got the second result

**Comment by frankfliu** [comment]
@mxnet-label-bot add [ndarray, bug]
